From 1b2925ea622d0bbd6a876b1096d8bd39f06e26ab Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Tue, 25 Nov 2014 16:33:59 -0500
Subject: [fs] bad_inode: add ->rename2()

Message-id: <20141125163359.18115.88295.stgit@warthog.procyon.org.uk>
Patchwork-id: 100916
O-Subject: [PATCH 15/47] bad_inode: add ->rename2()
Bugzilla: 985875
RH-Acked-by: Steve Best <sbest@redhat.com>

From: Miklos Szeredi <mszeredi@suse.cz>

so we return -EIO instead of -EINVAL.

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Original-commit: a0dbc56610b3e157f19241404e738744b7e7877e
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/bad_inode.c b/fs/bad_inode.c
index 922ad46..c8919ea 100644
--- a/fs/bad_inode.c
+++ b/fs/bad_inode.c
@@ -218,8 +218,9 @@ static int bad_inode_mknod (struct inode *dir, struct dentry *dentry,
  return -EIO;
 }
 
-static int bad_inode_rename (struct inode *old_dir, struct dentry *old_dentry,
-  struct inode *new_dir, struct dentry *new_dentry)
+static int bad_inode_rename2(struct inode *old_dir, struct dentry *old_dentry,
+        struct inode *new_dir, struct dentry *new_dentry,
+        unsigned int flags)
 {
  return -EIO;
 }
@@ -279,7 +280,7 @@ static const struct inode_operations bad_inode_ops =
  .mkdir  = bad_inode_mkdir,
  .rmdir  = bad_inode_rmdir,
  .mknod  = bad_inode_mknod,
- .rename  = bad_inode_rename,
+ .rename2 = bad_inode_rename2,
  .readlink = bad_inode_readlink,
  /* follow_link must be no-op, otherwise unmounting this inode
     won't work */
-- 
1.7.1